core/types/transaction_signing.go

```diff
@@ -170,7 +170,7 @@ type Signer interface {
 	Equal(Signer) bool
 
 	// Unwrap returns unwrapped signer implementation
-	Unwrap() interface{}
+	Unwrap() Signer
```

Comment:
Would it be simpler with something like the below instead of modifying the interfaces?

func Unwrap(s Signer) Signer {
    cs, ok := s.(CachedSigner);
    return (ok) ? cs.Signer : s;
}